Dalamud/DiscordBot/DiscordBotManager.cs

```diff
@@ -315,12 +348,187 @@ public async Task ProcessCfPreferredRoleChange(string rouletteName, string prevR
+                {"\uE0b2", "❷" }, // 
+                {"\uE0b3", "❸" }, // 
+                {"\uE0b4", "❹" }, // 
+                {"\uE0b5", "❺" }, // 
+                {"\uE0b6", "❻" }, // 
+                {"\uE0b7", "❼" }, // 
+                {"\uE0b8", "❽" }, // 
+                {"\uE0b9", "❾" }, // 
+                {"\uE0bb", "➲"}, //linked object arrow
+                {"\uE0bc", "☯️" }, // level sync icon
+                {"\uE0bd", "♋️" }, // level sync icon reversed
+                {"\uE0be", "🔽" }, // sync down icon
+                {"\uE0bf", "☒" }, // x in diamond
+
+                {"\uE0c0", "🌟" }, // wide star
+                {"\uE0c1", "Ⅰ" }, // wide roman numeral I
+                {"\uE0c2", "Ⅱ" }, // wide roman numeral II
+                {"\uE0c3", "Ⅲ" }, // wide roman numeral III
+                {"\uE0c4", "Ⅳ" }, // wide roman numeral IV
+                {"\uE0c5", "Ⅴ" }, // wide roman numeral V
+                {"\uE0c6", "Ⅵ" }, // wide roman numeral VI
+
+                {"\uE0D0", "Lᴛ" }, //LocalTimeEn   
+                {"\uE0D1", "Sᴛ" }, //ServerTimeEn   
+                {"\uE0D2", "Eᴛ" }, //EorzeaTimeEn   
+                {"\uE0D3", "Oᴢ" }, //LocalTimeDe   
+                {"\uE0D4", "Sᴢ" }, //ServerTimeDe   
+                {"\uE0D5", "Eᴢ" }, //EorzeaTimeDe   
+                {"\uE0D6", "Hʟ" }, //LocalTimeFr   
+                {"\uE0D7", "Hs" }, //ServerTimeFr   
+                {"\uE0D8", "Hᴇ" }, //EorzeaTimeFr   
+                {"\uE0D9", "本" }, //LocalTimeJa   
+                {"\uE0DA", "服" }, //ServerTimeJa   
+                {"\uE0DB", "艾" }, //EorzeaTimeJa   
+            };
+            // Transform the block letters into regional indicator letters. Will combine with flags if they are together in the right combo, so also adding a zero-width space
+            for( var i = 0; i < 26; i++ )
+            {
+                char xivchar = (char)(0xe071 + i);
+                string unichar = char.ConvertFromUtf32('A' + 0x1f1a5 + i);
+                string zerowidth = "\u200B";
+
+                ffxivSpecialCharacters.Add(xivchar.ToString(), unichar + zerowidth);
+            }
+            
+            // Number squares changed to enclosed number unicode. ⓪ is special, then ①-⑳ are one sequence then ㉑-㊿ are yet another, however xiv only goes to 31
+            for ( var i = 0; i <= 31; i++ )
+            {
+                char xivchar = (char)(0xe08f + i);
+                string unichar = "";
+                if( i == 0 )
+                {
+                    unichar = char.ConvertFromUtf32('⓪' + i);
+                }
+                if( i > 0 && i <= 19 )
+                {
+                    unichar = char.ConvertFromUtf32('①' + i - 1);
+                }
+                if( i >= 21 )
+                {
+                    unichar = char.ConvertFromUtf32('㉑' + i - 21);
+                }
+
+                ffxivSpecialCharacters.Add(xivchar.ToString(), unichar);
+            }
```

Comment:
Can you pull this out into a one-time method, rather than for every message?  It's basically a const dictionary, there's no reason to do this work every time.

Also for goat - a cleaned up version of this might be useful higher up in dalamud, though now that we have the game symbol and font handling, it may not be used much.

Reply:
I will update this to not build it every call but this is nowhere close to being called frequently enough to where this will impact literally anything